File: design-patterns/go/builder/request/request.go
// request.go

package main

import "fmt"

type RequestType int

const (
	GET RequestType = iota
	POST
	PUT
	PATCH
	DELETE
)

type Request struct {
	url         string
	requestType RequestType
	header      map[string]string
	body        map[string]string

	Builder
}

func (rcvr *Request) Send() {
	fmt.Println("Sending Request...")
	fmt.Printf("\nURL: %v", rcvr.url)
	fmt.Printf("\nRequest Type: %v", rcvr.requestType)
	fmt.Printf("\nHeaders: %v", rcvr.header)
	fmt.Printf("\nBody: %v", rcvr.body)

	// Write code for actually sending the request here
}

File: design-patterns/php/builder/request/Request.php
<?php
// Request.php

namespace BigBoxCode\DesignPattern\Builder\Request;

use BigBoxCode\DesignPattern\Builder\Request\Builder;

enum RequestType {
    case GET;
    case POST;
    case PUT;
    case PATCH;
    case DELETE;
}

class Request {
    private string $url;
    private RequestType $type;
    private array $header = [];
    private array $body = [];

    public function __construct(Builder $builder) {
        $this->url = $builder->url;
        $this->type = $builder->type;
        $this->header = $builder->header;
        $this->body = $builder->body;
    }

    public function send(): void {
        echo "Sending Request...";
        echo "\n\nURL: " . $this->url;
        echo "\n\nType: " . print_r($this->type, true);
        echo "\n\nHeaders: " . print_r($this->header, true);
        echo "\n\nBody: " . print_r($this->body, true);

        // Write functional code to send request
    }

    public static function newBuilder(): Builder {
        return new Builder();
    }
}